src/pieces/commandManager.ts

```diff
@@ -11,8 +13,51 @@ async function register(bot: Client): Promise<void> {
 		bot.emit('error', error);
 	}
 
-	bot.on('interactionCreate', interaction => {
+	bot.on('interactionCreate', async interaction => {
 		if (interaction.isCommand()) runCommand(interaction, bot);
+		if (interaction.isSelectMenu()) {
+			const courses: Array<Course> = await interaction.client.mongo.collection(DB.COURSES).find().toArray();
+			const { customId, values, member } = interaction;
+			let responseContent = `Your roles have been updated.`;
+			if (customId === 'roleselect' && member instanceof GuildMember) {
+				const component = interaction.component as MessageSelectMenu;
+				const removed = component.options.filter((option) => !values.includes(option.value));
+				for (const id of removed) {
+					const role = interaction.guild.roles.cache.find(r => r.id === id.value);
+					if (!role.name.includes('CISC')) {
+						member.roles.remove(id.value);
+						continue;
+					}
+					if (member.roles.cache.some(r => r.id === id.value)) { // does user have this role?
+						const course = courses.find(c => c.name === role.name.substring(5));
+						const user: SageUser = await interaction.client.mongo.collection(DB.USERS).findOne({ discordId: member.id });
+						user.courses = user.courses.filter(c => c !== course.name);
+						member.roles.remove(course.roles.student, `Unenrolled from ${course.name}.`);
+						member.roles.remove(id.value);
+						interaction.client.mongo.collection(DB.USERS).updateOne({ discordId: member.id }, { $set: { ...user } });
+						responseContent = `Your enrollments have been updated.`;
+					}
+				}
+				for (const id of values) {
```

Comment:
Ren and I were messing around with converting this to a forEach loop the other day to no avail, mainly because the `continue` statement doesn't work in such a loop. We've tried different things, including replacing it with a `return` but problems kept popping up, e.g. if a user selected two roles, only one would be assigned to them